great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -1038,3 +1076,46 @@ def add_query_asset(
             batch_metadata=batch_metadata or {},
         )
         return self._add_asset(asset)
+
+
+def _verify_table_name_exists_and_get_normalized_typed_name_map(
+    name: str,
+    typed_names: List[str | sqlalchemy.quoted_name],
+    error_message_template: str = 'Error: The table "{table_name:s}" does not exist.',
+    verify_only: bool = False,
+) -> Tuple[str, str | sqlalchemy.quoted_name] | None:
+    """
+    Verifies that table name exists.
+
+    Args:
+        name: string-valued name
+        typed_names: Properly typed names (e.g., output of "inspector.get_table_names()" SQLAlchemy method call)
+        verify_only: Perform verification only (do not return normalized typed name map)
+        error_message_template: String template to output error message if name cannot be found in typed_names list
+
+    Returns:
+        Single tuple holding mapping from string-valued name to typed name; None if "verify_only" is set.
```

Comment:
I'd rather we avoid union returns - if the caller wants to ignore the return value, they can do so.

Reply:
@cdkini this has been cleaned up / committed -- thanks.